mooncake-common/tests/CMakeLists.txt

```diff
@@ -1,6 +1,7 @@
 add_executable(default_config_test default_config_test.cpp)
 target_link_libraries(default_config_test PUBLIC
     mooncake_common 
+    Threads::Threads
```

Comment:
I suggest replacing Thread::Threads with pthread to avoid adding a new FindThreads.cmake dependency.